### ISSUE TITLE
Uncaught Error: Connection got disposed

### ISSUE THREAD

**Issue by soumesh2002** [redhat-developer/vscode-java#2268]
[provide a description of the issue]

##### Environment
- Operating System: Windows 11
- JDK version: 17.0.1
- Visual Studio Code version: 1.63.2
- Java extension version: 1.2.0

<img width="551" alt="image" src="https://user-images.githubusercontent.com/53593544/147830155-99d0f53a-c2b5-4b78-bc99-823737c47fed.png">

**Comment by jo-zzy** [comment]
I get the same error

> Environment

- Operating System: Windows 10
- JDK version: 11.0.13
- Visual Studio Code version: 1.63.2
- Java extension version: 1.2.0

**Comment by KyL0N** [comment]
same error :(

**Comment by testforstephen** [comment]
Could someone share the logs from these places?

- **F1 -> Java: Open All Log Files**
- **Help -> Toggle Developer Tools -> Console**

**Comment by AdityaSre** [comment]
Same error.

**Comment by testforstephen** [comment]
I also see some uncaught errors in the Runtime Status page when opening a Java file in VS Code. However, it seems the Java extension still works well, code completion, go to definition, debugging and other features all work.

**Comment by soumesh2002** [comment]
@testforstephen yes.. you're right everything works fine but there is an ⚠️error message showing as.
This extension got an uncaught error which is redirecting to runtime status of the extension and saying connection got disposed. I think this is a bug in the extension which is been there for some time now and not been fixed😁

**Comment by SystemOut1324** [comment]
I have a problem with snippets from server.json not being available do you have the same?

**Comment by soumesh2002** [comment]
Is there any fix available for above 'connection got disposed' error😁 @testforstephen 
Thank You

**Comment by testforstephen** [comment]
what I saw on this issue is just some logging issue. If you find any feature broken, could anyone provide more details about your environment and reproduce steps? thanks.

**Comment by michalhosala** [comment]
@testforstephen for me navigation through project stops working when I observe this error, i.e. I place a cursor on a symbol, choose "go to definition" or similar and it does nothing, it cannot navigate.

It doesnt happen straight away when opening project, navigation works fine for a while, but f I leave it open for like hour or so it always happens...

Language Support for Java(TM) by Red Hat: 1.4.0
Version: 1.65.2 (user setup)
Commit: c722ca6c7eed3d7987c0d5c3df5c45f6b15e77d1
Date: 2022-03-10T14:33:55.248Z (1 wk ago)
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19043

**Comment by testforstephen** [comment]
@michalhosala This error message is an expected behavior, your issue should be a different issue.

The Java extension will start a lightweight server and a standard server. When standard server is ready, we will shutdown  lightweight server because standard server is able to provide full feature set of lightweight server. The error message "Uncaught Error: Connection got disposed" occurs when lightweight server is killed. This is expected.

Could you pls open a new issue for your case? It's better to provide a minimum sample project to help reproduce your issue. Meanwhile, would you mind sharing your logs at "**F1** -> **Java: Open All Log Files**"? thanks.

**Comment by michalhosala** [comment]
@testforstephen ah, I see, got things mixed up... Unfortunately I won' be of much help, cause I returned my laptop just today as I am quitting my job, so no exact repro for you, sorry. However, I did have a look at logs and when I put a cursor on e.g. `Set<string>` and choos "go to definition" from context menu, then this is the last entry in the logs:
```
!ENTRY org.eclipse.jdt.ls.core 1 0 2022-03-29 23:46:10.848
!MESSAGE >> document/definition
```
and it is just stuck forever... In "Language support for Java (syntax server)" output window I see:
```
[Error - 3:30:00 PM] Connection to server got closed. Server will not be restarted.
```
Dont know if it is related.... Sorry for not being of more help here.

**Comment by ThibaultDelaune-pro** [comment]
Hi, I have the same error with redhat.java extension. In my case, it prevents the use of the Concourse VSCode extension.

Here is the Java logs
[log.txt](https://github.com/redhat-developer/vscode-java/files/10277317/log.txt)

I managed to make it temporarily work in installing the redhat.java pre-release version but after reboot the issue came back.

**Comment by nos-nart** [comment]
I got the same error on WSL2, JDK 19.0.1, and extension v1.13.0

**Comment by leorue05** [comment]
the problem still exists, i get this same message, i've tried a lot of things, like restarting my computer, re-installing the language support, even uninstalling and reinstalling vscode, nothing works, i've been suufering for 3 hours ,may break my hand on a wall if i keep getting frustrated
